uc-crux-llvm/src/UCCrux/LLVM/Run/Simulate.hs

```diff
@@ -149,17 +149,12 @@ symCreateOverrideFn ::
   SymCreateOverrideFn sym arch
 symCreateOverrideFn ov = SymCreateOverrideFn $ runCreateOverrideFn ov
 
--- NOTE(lb): The explicit kind signature here is necessary for GHC 8.6
--- compatibility.
 data UCCruxSimulationResult m arch (argTypes :: Ctx (FullType m)) = UCCruxSimulationResult
   { unsoundness :: Unsoundness,
     explanations :: [Located (Explanation m arch argTypes)]
   }
 
 -- | Based on 'Crux.SimulatorHooks'
---
--- NOTE(lb): The explicit kind signature here is necessary for GHC 8.6
--- compatibility.
 data SimulatorHooks sym m arch (argTypes :: Ctx (FullType m)) r =
```

Comment:
Might as well remove the kind signature if it's no longer needed. (Same comment applies to other places where NOTEs are removed)

Reply:
Sounds good to me. I didn't want to apply this change unilaterally since it comes down to a matter of preference, but I'll happily do so now that you've made your preference clear :)

Reply:
I appreciate the consideration; I think we should either kind signature everything or nothing for the sake of consistency, and the majority of things lack kind signatures at the moment.